### PR TITLE
compile.py: return error when test is absent

### PR DESCRIPTION
**Pull request by eroussy**
Print a log if a test is absent.
Return error in that case.

### CODE REVIEW

File: compile.py [hunk 1]
```diff
@@ -299,7 +304,9 @@ def add_compliance_matrix(xml_files, adoc_file, has_test_id):
                 requirements = list(sorted(csv.reader(matrix_file)))
                 # requirements is a list, each item of the list has the form
                 # ["requirement name", test_ID]
-                write_matrix_tests(requirements, xml_files, adoc_file)
+                ret = write_matrix_tests(requirements, xml_files, adoc_file)
+                if ret == 1:
```
Comment (kevlhop) [inline compile.py:308]: why don't you do directly ?

> return_code =  write_matrix_tests(requirements, xml_files, adoc_file) 

As the function write_matrix_tests is returning 0 or 1

File: compile.py [hunk 2]
```diff
@@ -358,6 +369,8 @@ def write_matrix_tests(requirements, xml_files, adoc_file):
                     _color_=ORANGE_COLOR,
                 )
             )
+            print(f"Test id {test_id} is not present")
```
Comment (kevlhop) [inline compile.py:372]: As there is no logging level, you could maybe put "ERROR:" before the message as this is returning an error

**Review by kevlhop — CHANGES_REQUESTED** [review]

File: compile.py (hunk 1 above)
Comment (eroussy) [reply compile.py:308]: write matrix is called multiple time.
Your line of code would take the return code of the last call.
We want it to return 1 if one of the call to write_matrix is 1

File: compile.py (hunk 2 above)
Comment (eroussy) [reply compile.py]: done

**Review by kevlhop — APPROVED** [review]

**Pull request merged**